conda_smithy/lint_recipe.py

```diff
@@ -64,7 +64,12 @@ def lintify(meta, recipe_dir=None):
         if bad_selectors:
             lints.append('Selectors are suggested to take a "  # [<selector>]" '
                          'form.')
-    
+
+    # 7: The build section should have a build number.
+    build_number = meta.get('build', {}).get('number', None)
```

Comment:
In order to be more resilient to this kind of thing:

```
package:
   name: foo

build:
   skip: true  # [False]

```

I try to put the item into a dictionary as per: https://github.com/ocefpaf/conda-smithy/blob/ensure_build_number/conda_smithy/lint_recipe.py#L31

That means we can deal with what yaml perceives as lists, but we as humans perceive as dicts.